examples/ui/drag_drop.rs

```diff
@@ -0,0 +1,374 @@
+                        "Drop into free space!".to_string(),
+                        TextStyle {
+                            font: asset_server.load("fonts/FiraMono-Medium.ttf"),
+                            font_size: 20.,
+                            color: Color::GRAY,
+                        },
+                    ));
+                });
+        });
+}
+
+fn mouse_hover(
+    mut event_mouse_enter: EventWriter<MouseEnterEvent>,
+    mut query_start_hover: Query<(Entity, &Interaction), (Changed<Interaction>, Without<MouseIn>)>,
+    mut event_mouse_exit: EventWriter<MouseExitEvent>,
+    mut query_stop_hover: Query<(Entity, &Interaction), (Changed<Interaction>, With<MouseIn>)>,
+) {
+    for (entity, interaction) in query_start_hover.iter_mut() {
+        match *interaction {
+            Interaction::Clicked | Interaction::None => {}
+            Interaction::Hovered => {
+                event_mouse_enter.send(MouseEnterEvent(entity));
+            }
+        }
+    }
+
+    for (entity, interaction) in query_stop_hover.iter_mut() {
+        match *interaction {
+            Interaction::Clicked | Interaction::Hovered => {}
+            Interaction::None => {
+                event_mouse_exit.send(MouseExitEvent(entity));
+            }
+        }
+    }
+}
+
+fn mouse_enter(mut mouse_enter_event: EventReader<MouseEnterEvent>, mut commands: Commands) {
+    for mouse_enter in mouse_enter_event.iter() {
+        commands.entity(mouse_enter.0).insert(MouseIn);
+    }
+}
+
+fn mouse_exit(mut mouse_exit_event: EventReader<MouseExitEvent>, mut commands: Commands) {
+    for mouse_exit in mouse_exit_event.iter() {
+        commands.entity(mouse_exit.0).remove::<MouseIn>();
+    }
+}
+
+fn mouse_drag_not_active(
+    mut query_list: Query<
+        (&Interaction, Entity),
+        (Changed<Interaction>, Without<DragActive>, With<Dragable>),
+    >,
+    mut event_hover_dragable: EventWriter<DragableHoverEvent>,
+    mut event_click_dragable: EventWriter<DragableClickEvent>,
+) {
+    for (interaction, entity) in query_list.iter_mut() {
+        match *interaction {
+            Interaction::Clicked => {
+                event_click_dragable.send(DragableClickEvent(entity));
+            }
+            Interaction::Hovered => {
+                event_hover_dragable.send(DragableHoverEvent(entity));
+            }
+            Interaction::None => {}
+        }
+    }
+}
+
+fn mouse_drag_active_moving(
+    res_window: Res<Windows>,
+    mouse_motion_events: EventReader<MouseMotion>,
+    mut query_list: Query<&mut Style, With<DragMoving>>,
+) {
+    if query_list.is_empty() || mouse_motion_events.is_empty() {
+        return;
+    }
+
+    if let Some(mouse_position) = res_window.get_primary().unwrap().cursor_position() {
+        for style in query_list.iter_mut() {
+            update_style_with_mouse_position(style, mouse_position);
+        }
+    }
+}
+
+fn update_style_with_mouse_position(mut style: Mut<Style>, position: Vec2) {
```

Comment:
I would probably reframe this as `update_position` in some form; `style` is a very overloaded term (and we're likely to split apart the component).

Reply:
ok